pkg/issuer/acme/dns/azuredns/azuredns.go

```diff
@@ -41,25 +43,31 @@ type DNSProvider struct {
 // NewDNSProviderCredentials returns a DNSProvider instance configured for the Azure
 // DNS service using static credentials from its parameters
 func NewDNSProviderCredentials(environment, clientID, clientSecret, subscriptionID, tenantID, resourceGroupName, zoneName string, dns01Nameservers []string, ambient bool, managedIdentity *cmacme.AzureManagedIdentity) (*DNSProvider, error) {
-	env := azure.PublicCloud
-	if environment != "" {
-		var err error
-		env, err = azure.EnvironmentFromName(environment)
-		if err != nil {
-			return nil, err
-		}
+  c := cloud.AzurePublic
+  if environment != "" {
```

Comment:
German cloud missing from this list.

Reply:
It looks like German cloud was closed in October 2021

> Additionally, on Sept 30th, 2020, we announced that the Microsoft Cloud Germany would be closing on October 29th, 2021

https://learn.microsoft.com/en-us/previous-versions/azure/germany/germany-welcome

So perhaps we need to trim this list of supported Azure cloud environments:

https://github.com/cert-manager/cert-manager/blob/8deaca755d54a7aee5cf1a3f68a7f952924f6302/pkg/apis/acme/v1/types_issuer.go#L550-L558